recipes/ssht/all/conanfile.py

```diff
@@ -13,47 +17,52 @@ class SshtConan(ConanFile):
     description = "Fast spin spherical harmonic transforms"
     settings = "os", "arch", "compiler", "build_type"
     topics = ("physics", "astrophysics", "radio interferometry")
+    package_type = "static-library"
     options = {"fPIC": [True, False]}
     default_options = {"fPIC": True}
-    requires = "fftw/3.3.9"
-    generators = "cmake", "cmake_find_package", "cmake_paths"
-    exports_sources = ["CMakeLists.txt"]
 
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
 
-    @property
-    def _build_subfolder(self):
-        return "build_subfolder"
+    def configure(self):
+        self.settings.rm_safe("compiler.libcxx")
+        self.settings.rm_safe("compiler.cppstd")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def requirements(self):
+        self.requires("fftw/3.3.9")
 
-    def config_options(self):
-        del self.settings.compiler.cppstd
-        del self.settings.compiler.libcxx
-    
     def validate(self):
-        if self.settings.compiler == "Visual Studio":
+        if is_msvc(self):
             raise ConanInvalidConfiguration("SSHT requires C99 support for complex numbers.")
 
     def source(self):
-        files.get(self, **self.conan_data["sources"][self.version],
-                  destination=self._source_subfolder, strip_root=True)
-
-    @property
-    def cmake(self):
-        if not hasattr(self, "_cmake"):
-            self._cmake = CMake(self)
-            self._cmake.definitions["tests"] = False
-            self._cmake.definitions["python"] = False
-            self._cmake.configure(build_folder=self._build_subfolder)
-        return self._cmake
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.cache_variables["tests"] = False
+        tc.cache_variables["python"] = False
+        tc.generate()
+        tc = CMakeDeps(self)
+        tc.generate()
+
 
     def build(self):
-        self.cmake.build()
+        replace_in_file(self, os.path.join(self.source_folder, "src", "c", "CMakeLists.txt"),
```

Comment:
I think we could do: 
https://docs.conan.io/2/reference/tools/cmake/cmakedeps.html#overwrite-properties-from-the-consumer-side-using-cmakedeps-set-property

in the `generate()` method to make the target appear as `FFTW3::FFTW3` and avoid patching the recipe